presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.h

```diff
@@ -169,4 +162,46 @@ class VeloxQueryPlanConverter {
   VeloxExprConverter exprConverter_;
 };
 
+class VeloxQueryPlanConverterImpl : public VeloxQueryPlanConverterBase {
+ public:
+  using VeloxQueryPlanConverterBase::toVeloxQueryPlan;
+
+  explicit VeloxQueryPlanConverterImpl(velox::memory::MemoryPool* pool)
+      : VeloxQueryPlanConverterBase(pool) {}
+
+ protected:
+  velox::core::PlanNodePtr toVeloxQueryPlan(
+      const std::shared_ptr<const protocol::RemoteSourceNode>& node,
+      const std::shared_ptr<protocol::TableWriteInfo>& tableWriteInfo,
+      const protocol::TaskId& taskId) override;
+};
+
+class VeloxQueryPlanConverterBatch : public VeloxQueryPlanConverterBase {
+ public:
+  using VeloxQueryPlanConverterBase::toVeloxQueryPlan;
+
+  VeloxQueryPlanConverterBatch(
+      const std::string& shuffleName,
+      std::shared_ptr<std::string>&& serializedShuffleWriteInfo,
+      velox::memory::MemoryPool* pool)
+      : VeloxQueryPlanConverterBase(pool),
+        shuffleName_(shuffleName),
+        serializedShuffleWriteInfo_(std::move(serializedShuffleWriteInfo)) {}
+
+  velox::core::PlanFragment toVeloxQueryPlan(
+      const protocol::PlanFragment& fragment,
+      const std::shared_ptr<protocol::TableWriteInfo>& tableWriteInfo,
+      const protocol::TaskId& taskId) override;
+
+ protected:
+  velox::core::PlanNodePtr toVeloxQueryPlan(
+      const std::shared_ptr<const protocol::RemoteSourceNode>& node,
+      const std::shared_ptr<protocol::TableWriteInfo>& tableWriteInfo,
+      const protocol::TaskId& taskId) override;
+
+ private:
+  std::string shuffleName_;
```

Comment:
Both shuffleName_ and serializedShuffleWriteInfo_ are consts?

Reply:
A const member indicates the value never changes during an object's lifecycle. It might help code reading and debugging. If we need to move a class object, we could either use unique_ptr for that, or design it as a data class (which doesn't expose any public API for mutation after the object creation) in case of trivial class.